In [1]:
# Standard modules
import os
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

## Read in the data and take a quick look at it

In [2]:
# Read in the merged dataset
csv_filename = '/Users/jmason86/Dropbox/Research/Postdoc_NASA/Analysis/Coronal Dimming Analysis/JEDI Catalog/jedi_merged_2018-05-03 18:20:27.469.csv'
jedi_catalog_df = pd.read_csv(csv_filename, low_memory=False)

In [6]:
print(jedi_catalog_df.shape)
jedi_catalog_df.head()

(620, 24304)


Event #    GOES Flare Start Time     GOES Flare Peak Time GOES Flare Class  \
0      9.0  2010-06-12 00:30:00.000  2010-06-12 00:57:00.000             M2.0   
1     10.0  2010-06-12 03:57:00.000  2010-06-12 04:06:00.000             C1.0   
2     11.0  2010-06-12 09:02:00.000  2010-06-12 09:17:00.000             C6.1   
3     12.0  2010-06-13 05:30:00.000  2010-06-13 05:39:00.000             M1.0   
4     13.0  2010-06-13 06:08:00.000  2010-06-13 06:11:00.000             C1.2   

      Pre-Flare Start Time       Pre-Flare End Time  Flare Interrupt  \
0  2010-06-11 16:57:00.000  2010-06-12 00:57:00.000             True   
1  2010-06-11 16:57:00.000  2010-06-12 00:57:00.000             True   
2  2010-06-11 16:57:00.000  2010-06-12 00:57:00.000             True   
3  2010-06-12 21:39:00.000  2010-06-13 05:39:00.000             True   
4  2010-06-12 21:39:00.000  2010-06-13 05:39:00.000             True   

   9.4 Pre-Flare Irradiance [W/m2]  13.1 Pre-Flare Irradiance [W/m2]  \
0                              NaN                          0.000002   
1                              NaN                          0.000002   
2                              NaN                          0.000002   
3                              NaN                               NaN   
4                              NaN                               NaN   

   13.3 Pre-Flare Irradiance [W/m2]              ...               \
0                               NaN              ...                
1                               NaN              ...                
2                               NaN              ...                
3                               NaN              ...                
4                               NaN              ...                

   103.2 by 63.0 Fitting Score  103.2 by 71.9 Fitting Score  \
0                          NaN                          NaN   
1                          NaN                          NaN   
2                          NaN                          NaN   
3                          NaN                          NaN   
4                          NaN                          NaN   

   103.2 by 72.2 Fitting Score  103.2 by 77.0 Fitting Score  \
0                          NaN                          NaN   
1                          NaN                          NaN   
2                          NaN                          NaN   
3                          NaN                          NaN   
4                          NaN                          NaN   

   103.2 by 79.0 Fitting Score  103.2 by 83.6 Fitting Score  \
0                          NaN                          NaN   
1                          NaN                          NaN   
2                          NaN                          NaN   
3                          NaN                          NaN   
4                          NaN                          NaN   

   103.2 by 95.0 Fitting Score  103.2 by 97.3 Fitting Score  \
0                          NaN                          NaN   
1                          NaN                          NaN   
2                          NaN                          NaN   
3                          NaN                          NaN   
4                          NaN                          NaN   

   103.2 by 97.7 Fitting Score  103.2 by 102.6 Fitting Score  
0                          NaN                           NaN  
1                          NaN                           NaN  
2                          NaN                           NaN  
3                          NaN                           NaN  
4                          NaN                           NaN  

[5 rows x 24304 columns]

## New dataframe of just Dave and Alysha's events

In [21]:
event_numbers = [30, 93, 142, 576, 589, 590, 673, 682, 785, 810, 901, 954, 1477, 1487, 1535, 1535, 
                 1552, 1588, 1589, 1617, 1968, 2025, 2128, 2143, 2244, 3040, 3568, 3701]
jedi_sub_catalog_df = jedi_catalog_df[jedi_catalog_df['Event #'].isin(event_numbers)]
jedi_sub_catalog_df

Event #    GOES Flare Start Time     GOES Flare Peak Time  \
21      30.0  2010-08-01 07:55:00.000  2010-08-01 08:26:00.000   
113     93.0  2010-12-14 15:03:00.000  2010-12-14 15:50:00.000   
162    142.0  2011-02-15 01:44:00.000  2011-02-15 01:56:00.000   
418    576.0  2011-06-02 07:22:00.000  2011-06-02 07:46:00.000   
431    589.0  2011-06-14 00:32:00.000  2011-06-14 00:41:00.000   
432    590.0  2011-06-14 11:13:00.000  2011-06-14 11:29:00.000   
488    673.0  2011-08-02 09:24:00.000  2011-08-02 09:27:00.000   
497    682.0  2011-08-03 13:17:00.000  2011-08-03 13:48:00.000   
598    785.0  2011-09-06 22:12:00.000  2011-09-06 22:20:00.000   
599    785.0  2011-09-06 22:12:00.000  2011-09-06 22:20:00.000   
602    810.0  2011-09-13 23:31:00.000  2011-09-13 23:36:00.000   
603    901.0  2011-09-24 12:33:00.000  2011-09-24 13:20:00.000   
604    954.0  2011-10-02 00:37:00.000  2011-10-02 00:50:00.000   
605   1477.0  2012-01-19 12:41:00.000  2012-01-19 12:50:00.000   
606   1487.0  2012-01-23 03:38:00.000  2012-01-23 03:59:00.000   
607   1535.0  2012-02-26 11:19:00.000  2012-02-26 11:25:00.000   
608   1552.0  2012-03-05 02:30:00.000  2012-03-05 04:09:00.000   
609   1588.0  2012-03-07 00:02:00.000  2012-03-07 00:24:00.000   
610   1589.0  2012-03-07 01:05:00.000  2012-03-07 01:14:00.000   
611   1617.0  2012-03-10 17:15:00.000  2012-03-10 17:44:00.000   
612   1968.0  2012-06-08 02:51:00.000  2012-06-08 03:07:00.000   
613   2025.0  2012-06-14 12:33:00.000  2012-06-14 12:37:00.000   
614   2128.0  2012-07-03 20:36:00.000  2012-07-03 20:45:00.000   
615   2143.0  2012-07-04 16:33:00.000  2012-07-04 16:39:00.000   
616   2244.0  2012-07-12 15:37:00.000  2012-07-12 16:49:00.000   
617   3040.0  2013-03-15 05:46:00.000  2013-03-15 06:58:00.000   
618   3568.0  2013-07-09 13:25:00.000  2013-07-09 13:32:00.000   
619   3701.0  2013-09-29 21:43:00.000  2013-09-29 23:39:00.000   

    GOES Flare Class     Pre-Flare Start Time       Pre-Flare End Time  \
21              C3.2  2010-08-01 00:26:00.000  2010-08-01 08:26:00.000   
113             C2.3  2010-12-14 07:50:00.000  2010-12-14 15:50:00.000   
162             X2.2  2011-02-12 23:29:00.000  2011-02-13 07:29:00.000   
418             C3.7  2011-06-01 18:44:00.000  2011-06-02 02:44:00.000   
431             C1.8  2011-06-13 16:41:00.000  2011-06-14 00:41:00.000   
432             C2.2  2011-06-14 03:29:00.000  2011-06-14 11:29:00.000   
488             C2.0  2011-07-31 10:01:00.000  2011-07-31 18:01:00.000   
497             M6.0  2011-08-02 18:38:00.000  2011-08-03 02:38:00.000   
598             X2.1  2011-09-06 14:20:00.000  2011-09-06 22:20:00.000   
599             X2.1  2011-09-06 14:20:00.000  2011-09-06 22:20:00.000   
602             C2.9  2011-09-06 14:20:00.000  2011-09-06 22:20:00.000   
603             M7.1  2011-09-06 14:20:00.000  2011-09-06 22:20:00.000   
604             M3.9  2011-10-01 16:50:00.000  2011-10-02 00:50:00.000   
605             C3.2  2012-01-19 04:50:00.000  2012-01-19 12:50:00.000   
606             M8.7  2012-01-19 04:50:00.000  2012-01-19 12:50:00.000   
607             C1.3  2012-02-26 03:25:00.000  2012-02-26 11:25:00.000   
608             X1.1  2012-02-26 03:25:00.000  2012-02-26 11:25:00.000   
609             X5.4  2012-02-26 03:25:00.000  2012-02-26 11:25:00.000   
610             X1.3  2012-02-26 03:25:00.000  2012-02-26 11:25:00.000   
611             M8.4  2012-02-26 03:25:00.000  2012-02-26 11:25:00.000   
612             C7.7  2012-02-26 03:25:00.000  2012-02-26 11:25:00.000   
613             C2.6  2012-02-26 03:25:00.000  2012-02-26 11:25:00.000   
614             C9.3  2012-02-26 03:25:00.000  2012-02-26 11:25:00.000   
615             M1.8  2012-02-26 03:25:00.000  2012-02-26 11:25:00.000   
616             X1.4  2012-07-12 08:49:00.000  2012-07-12 16:49:00.000   
617             M1.1  2013-03-14 22:58:00.000  2013-03-15 06:58:00.000   
618             C2.3  2013-03-14 22:58:00.000  2013-03-15 06:58:00.000   
619             C1

## Start manually subselecting to get numbers to add to spreadhseet

In [43]:
pd.set_option('display.max_columns', 500)

Done with indices: 21, 113, 162, 432, 497, 604, 608, 610, 616, 618
Keep an eye on index 431 for a higher depth

In [128]:
cols = [col for col in jedi_sub_catalog_df.columns if '49.9' in col and 'Depth [%]' in col]
jedi_sub_catalog_df[cols].dropna(how='all')

49.9 Depth [%]  9.4 by 49.9 Depth [%]  13.1 by 49.9 Depth [%]  \
162             NaN                    NaN                 3.26309   
618          7.3496                    NaN                     NaN   

     13.3 by 49.9 Depth [%]  17.1 by 49.9 Depth [%]  17.7 by 49.9 Depth [%]  \
162                     NaN                     NaN                3.476285   
618                     NaN                     NaN                     NaN   

     18.0 by 49.9 Depth [%]  19.5 by 49.9 Depth [%]  20.2 by 49.9 Depth [%]  \
162                     NaN                     NaN                     NaN   
618                     NaN                     NaN                     NaN   

     21.1 by 49.9 Depth [%]  25.6 by 49.9 Depth [%]  28.4 by 49.9 Depth [%]  \
162                     NaN                     NaN                     NaN   
618                     NaN                     NaN                     NaN   

     30.4 by 49.9 Depth [%]  33.5 by 49.9 Depth [%]  36.1 by 49.9 Depth [%]  \
162                     NaN                     NaN                     NaN   
618                     NaN                     NaN                     NaN   

     36.8 by 49.9 Depth [%]  44.6 by 49.9 Depth [%]  46.5 by 49.9 Depth [%]  \
162                     NaN                     NaN                     NaN   
618                     NaN                     NaN                     NaN   

     49.9 by 9.4 Depth [%]  49.9 by 13.1 Depth [%]  49.9 by 13.3 Depth [%]  \
162                    NaN                     NaN                     NaN   
618                    NaN                     NaN                     NaN   

     49.9 by 17.1 Depth [%]  49.9 by 17.7 Depth [%]  49.9 by 18.0 Depth [%]  \
162                     NaN                     NaN                     NaN   
618                     NaN                     NaN                     NaN   

     49.9 by 19.5 Depth [%]  49.9 by 20.2 Depth [%]  49.9 by 21.1 Depth [%]  \
162                     NaN                     NaN                     NaN   
618                     NaN                     NaN                     NaN   

     49.9 by 25.6 Depth [%]  49.9 by 28.4 Depth [%]  49.9 by 30.4 Depth [%]  \
162                     NaN                     NaN                     NaN   
618                     NaN                     NaN                     NaN   

     49.9 by 33.5 Depth [%]  49.9 by 36.1 Depth [%]  49.9 by 36.8 Depth [%]  \
162                     NaN                     NaN                     NaN   
618                     NaN                     NaN                     NaN   

     49.9 by 44.6 Depth [%]  49.9 by 46.5 Depth [%]  49.9 by 52.1 Depth [%]  \
162                     NaN                     NaN                     NaN   
618                     NaN                     NaN                     NaN   

     49.9 by 52.6 Depth [%]  49.9 by 53.7 Depth [%]  49.9 by 55.4 Depth [%]  \
162                     NaN                     NaN                     NaN   
618                     NaN                     NaN                     NaN   

     49.9 by 56.8 Depth [%]  49.9 by 58.4 Depth [%]  49.9 by 59.2 Depth [%]  \
162                     NaN                     NaN                     NaN   
618                     NaN                     NaN                     NaN   

     49.9 by 60.0 Depth [%]  49.9 by 61.0 Depth [%]  49.9 by 62.5 Depth [%]  \
162                     NaN                     NaN                     NaN   
618                     NaN                     NaN                     NaN   

     49.9 by 63.0 Depth [%]  49.9 by 71.9 Depth [%]  49.9 by 72.2 Depth [%]  \
162                     NaN                     NaN                     NaN   
618                     NaN                     NaN                     NaN   

     49.9 by 77.0 Depth [%]  49.9 by 79.0 Depth [%]  49.9 by 83.6 Depth [%]  \
162                     NaN                     NaN                     NaN   
618                     NaN                     NaN                  

In [120]:
jedi_sub_catalog_df[jedi_sub_catalog_df['Event #'] == 785.0]['21.1 Depth Time']

611    2012-03-10 18:35:14
Name: 21.1 Depth Time, dtype: object

In [117]:
cols = [col for col in jedi_sub_catalog_df.columns if '21.1' in col and 'Slope Mean' in col]
jedi_sub_catalog_df[cols].dropna(how='all')

21.1 Slope Mean [%/s]  9.4 by 21.1 Slope Mean [%/s]  \
21                0.000492                           NaN   
113               0.000220                           NaN   
432                    NaN                           NaN   
607                    NaN                           NaN   
608               0.000357                           NaN   
610               0.000595                           NaN   
611               0.000592                           NaN   

     13.1 by 21.1 Slope Mean [%/s]  13.3 by 21.1 Slope Mean [%/s]  \
21                             NaN                            NaN   
113                            NaN                            NaN   
432                       0.001209                            NaN   
607                       0.001044                            NaN   
608                            NaN                            NaN   
610                            NaN                            NaN   
611                            NaN                            NaN   

     17.1 by 21.1 Slope Mean [%/s]  17.7 by 21.1 Slope Mean [%/s]  \
21                             NaN                            NaN   
113                            NaN                            NaN   
432                            NaN                            NaN   
607                            NaN                            NaN   
608                            NaN                            NaN   
610                            NaN                            NaN   
611                            NaN                            NaN   

     18.0 by 21.1 Slope Mean [%/s]  19.5 by 21.1 Slope Mean [%/s]  \
21                             NaN                            NaN   
113                            NaN                            NaN   
432                       0.000167                            NaN   
607                            NaN                            NaN   
608                            NaN                            NaN   
610                            NaN                            NaN   
611                            NaN                            NaN   

     20.2 by 21.1 Slope Mean [%/s]  21.1 by 9.4 Slope Mean [%/s]  \
21                             NaN                           NaN   
113                            NaN                           NaN   
432                            NaN                           NaN   
607                            NaN                           NaN   
608                            NaN                           NaN   
610                            NaN                           NaN   
611                            NaN                           NaN   

     21.1 by 13.1 Slope Mean [%/s]  21.1 by 13.3 Slope Mean [%/s]  \
21                             NaN                            NaN   
113                            NaN                            NaN   
432                            NaN                            NaN   
607                       0.000109                            NaN   
608                            NaN                            NaN   
610                            NaN                            NaN   
611                            NaN                            NaN   

     21.1 by 17.1 Slope Mean [%/s]  21.1 by 17.7 Slope Mean [%/s]  \
21                             NaN                            NaN   
113                            NaN                            NaN   
432                            NaN                            NaN   
607                            NaN                            NaN   
608                            NaN                            NaN   
610                            NaN                            NaN   
611                            NaN                            NaN   

     21.1 by 18.0 Slope Mean [%/s]  21.1 by 19.5 Slope Mean [%/s]  \
21                             NaN                            NaN   
113                            NaN                            NaN   
432    

In [118]:
cols = [col for col in jedi_sub_catalog_df.columns if '21.1' in col and 'Duration' in col]
jedi_sub_catalog_df[cols].dropna(how='all')

21.1 Duration Start Time 21.1 Duration End Time  21.1 Duration [s]  \
113      2010-12-14 19:17:05    2010-12-15 02:26:05            25740.0   
432                      NaN                    NaN                NaN   
607                      NaN                    NaN                NaN   

    9.4 by 21.1 Duration Start Time 13.1 by 21.1 Duration Start Time  \
113                             NaN                              NaN   
432                             NaN                              NaN   
607                             NaN                              NaN   

     13.3 by 21.1 Duration Start Time  17.1 by 21.1 Duration Start Time  \
113                               NaN                               NaN   
432                               NaN                               NaN   
607                               NaN                               NaN   

    17.7 by 21.1 Duration Start Time 18.0 by 21.1 Duration Start Time  \
113                              NaN                              NaN   
432                              NaN                              NaN   
607                              NaN                              NaN   

    19.5 by 21.1 Duration Start Time  20.2 by 21.1 Duration Start Time  \
113                              NaN                               NaN   
432                              NaN                               NaN   
607                              NaN                               NaN   

     21.1 by 9.4 Duration Start Time 21.1 by 13.1 Duration Start Time  \
113                              NaN                              NaN   
432                              NaN                              NaN   
607                              NaN              2012-02-26 14:02:10   

     21.1 by 13.3 Duration Start Time  21.1 by 17.1 Duration Start Time  \
113                               NaN                               NaN   
432                               NaN                               NaN   
607                               NaN                               NaN   

    21.1 by 17.7 Duration Start Time 21.1 by 18.0 Duration Start Time  \
113                              NaN                              NaN   
432                              NaN                              NaN   
607                              NaN                              NaN   

     21.1 by 19.5 Duration Start Time 21.1 by 20.2 Duration Start Time  \
113                               NaN                              NaN   
432                               NaN                              NaN   
607                               NaN                              NaN   

    21.1 by 25.6 Duration Start Time 21.1 by 28.4 Duration Start Time  \
113                              NaN                              NaN   
432                              NaN                              NaN   
607                              NaN                              NaN   

    21.1 by 30.4 Duration Start Time  21.1 by 33.5 Duration Start Time  \
113                              NaN                               NaN   
432                              NaN                               NaN   
607                              NaN                               NaN   

     21.1 by 36.1 Duration Start Time 21.1 by 36.8 Duration Start Time  \
113                               NaN                              NaN   
432                               NaN                              NaN   
607                               NaN                              NaN   

     21.1 by 44.6 Duration Start Time  21.1 by 46.5 Duration Start Time  \
113                               NaN                               NaN   
432                               NaN                               NaN   
607                               NaN                               NaN   

     21.1 by 49.9 Duration Start Time  21.1 by 52.1 Duration Start Time  \
113                               NaN                               NaN   
432  

In [66]:
cols = [col for col in jedi_sub_catalog_df.columns if '17.1 by ' in col]
jedi_sub_catalog_df[cols].dropna(how='all')

17.1 by 9.4 Slope Start Time 17.1 by 13.1 Slope Start Time  \
605                          NaN           2012-01-19 12:50:00   
606                          NaN           2012-01-23 03:59:00   
616                          NaN                           NaN   
618                          NaN                           NaN   

     17.1 by 13.3 Slope Start Time 17.1 by 17.7 Slope Start Time  \
605                            NaN                           NaN   
606                            NaN                           NaN   
616                            NaN                           NaN   
618                            NaN                           NaN   

     17.1 by 18.0 Slope Start Time 17.1 by 19.5 Slope Start Time  \
605                            NaN                           NaN   
606                            NaN                           NaN   
616                            NaN                           NaN   
618                            NaN                           NaN   

    17.1 by 20.2 Slope Start Time  17.1 by 21.1 Slope Start Time  \
605                           NaN                            NaN   
606                           NaN                            NaN   
616                           NaN                            NaN   
618                           NaN                            NaN   

    17.1 by 25.6 Slope Start Time 17.1 by 28.4 Slope Start Time  \
605                           NaN                           NaN   
606                           NaN           2012-01-23 03:59:00   
616                           NaN                           NaN   
618                           NaN                           NaN   

    17.1 by 30.4 Slope Start Time 17.1 by 33.5 Slope Start Time  \
605                           NaN                           NaN   
606                           NaN                           NaN   
616                           NaN                           NaN   
618                           NaN                           NaN   

     17.1 by 36.1 Slope Start Time  17.1 by 36.8 Slope Start Time  \
605                            NaN                            NaN   
606                            NaN                            NaN   
616                            NaN                            NaN   
618                            NaN                            NaN   

     17.1 by 44.6 Slope Start Time  17.1 by 46.5 Slope Start Time  \
605                            NaN                            NaN   
606                            NaN                            NaN   
616                            NaN                            NaN   
618                            NaN                            NaN   

    17.1 by 49.9 Slope Start Time 17.1 by 52.1 Slope Start Time  \
605                           NaN                           NaN   
606                           NaN                           NaN   
616                           NaN                           NaN   
618                           NaN                           NaN   

     17.1 by 52.6 Slope Start Time  17.1 by 53.7 Slope Start Time  \
605                            NaN                            NaN   
606                            NaN                            NaN   
616                            NaN                            NaN   
618                            NaN                            NaN   

     17.1 by 55.4 Slope Start Time  17.1 by 56.8 Slope Start Time  \
605                            NaN                            NaN   
606                            NaN                            NaN   
616                            NaN                            NaN   
618                            NaN                            NaN   

    17.1 by 58.4 Slope Start Time  17.1 by 59.2 Slope Start Time  \
605                           NaN                            NaN   
606                           NaN                            NaN   
616                           NaN                     